examples/BlazorServerApp/MyTargetingContextAccessor.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.AspNetCore.Components.Authorization;
+using Microsoft.FeatureManagement.FeatureFilters;
+
+namespace BlazorServerApp
+{
+    public class MyTargetingContextAccessor : ITargetingContextAccessor
+    {
+        private readonly AuthenticationStateProvider _authenticationStateProvider;
+
+        public MyTargetingContextAccessor(AuthenticationStateProvider authenticationStateProvider)
+        {
+            _authenticationStateProvider = authenticationStateProvider ?? throw new ArgumentNullException(nameof(authenticationStateProvider));
+        }
+
+        public async ValueTask<TargetingContext> GetContextAsync()
+        {
+            AuthenticationState authState = await _authenticationStateProvider.GetAuthenticationStateAsync();
+
+            string username = authState.User.Identity.Name;
```

Comment:
Shouldn't there be some null checks, or is identity non-null even for unauthenticated requests?